homeassistant/components/improv_ble/config_flow.py

```diff
@@ -152,6 +179,10 @@ async def async_step_bluetooth_confirm(
         # mypy is not aware that we can't get here without having these set already
         assert self._discovery_info is not None
 
+        # if self._remove_bluetooth_callback:
+        #    self._remove_bluetooth_callback()
+        #    self._remove_bluetooth_callback = None
```

Comment:
Temporarily commented out to debug unexpected bluetooth disconnects

Reply:
The problem is fixed by https://github.com/home-assistant-libs/py-improv-ble-client/pull/16